cudax/test/hierarchy/testing_common.cuh

```diff
@@ -22,14 +22,32 @@ namespace cudax = cuda::experimental;
 
 #define CUDART(call) REQUIRE((call) == cudaSuccess)
 
-#undef NDEBUG
-#include <cassert>
+static void __device__ cudax_require_impl(
+  bool condition, const char* condition_text, const char* filename, unsigned int linenum, const char* funcname)
```

Comment:
Suggestion: I assume you want one function for your program and not one in each TU.
```suggestion
inline void __device__ cudax_require_impl(
  bool condition, const char* condition_text, const char* filename, unsigned int linenum, const char* funcname)
```